src/kv/map.h

```diff
@@ -674,5 +689,42 @@ namespace kv
       std::swap(rollback_counter, map->rollback_counter);
       std::swap(roll, map->roll);
     }
+
+    template <typename TView>
+    TView* create_view(Version version)
```

Comment:
`create_view` is now a template (so that `experimental::Map` can `create_view<experimental::TxView>`), and doesn't pretend to return an abstract type. The only caller is `tx.h`, which knows exactly what kind of view it expects - it casts from `AbstractTxView` immediately after calling. So we return a concrete (templated) type, and then confirm that's castable to the `AbstractTxView` we want to store.